src/components/RequireAuth/RequireAuth.jsx

```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { useAuth } from "context";
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+
+export function RequireAuth() {
+  const {
+    auth: { encodedToken, isAuthenticated },
+  } = useAuth();
+  const location = useLocation();
+
+  if (encodedToken && isAuthenticated) return <Outlet />;
+  return <Navigate to='/login' state={{ from: location }} replace></Navigate>;
```

Comment:
suggestion: You could just give a closing tag to Navigate

Reply:
sure will do it.